Core/SqlPreparation/MethodCallTransformers/DateOnlyAddMethodCallTransformer.cs

```diff
@@ -0,0 +1,74 @@
+// This file is part of the re-linq project (relinq.codeplex.com)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// re-linq is free software; you can redistribute it and/or modify it under
+// the terms of the GNU Lesser General Public License as published by the
+// Free Software Foundation; either version 2.1 of the License,
+// or (at your option) any later version.
+//
+// re-linq is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-linq; if not, see http://www.gnu.org/licenses.
+//
+using System;
+using System.Linq.Expressions;
+using System.Reflection;
+using Remotion.Linq.SqlBackend.SqlGeneration;
+using Remotion.Linq.SqlBackend.SqlStatementModel.SqlSpecificExpressions;
+using Remotion.Utilities;
+
+#if NET6_0_OR_GREATER
+namespace Remotion.Linq.SqlBackend.SqlPreparation.MethodCallTransformers
+{
+  public class DateOnlyAddMethodCallTransformer : IMethodCallTransformer
+  {
+    public static readonly MethodInfo[] SupportedMethods = new[]
+      {
+        MethodCallTransformerUtility.GetInstanceMethod (
+            typeof (DateOnly),
+            "AddDays",
+            typeof (int)),
+        MethodCallTransformerUtility.GetInstanceMethod (
+            typeof (DateOnly),
+            "AddMonths",
+            typeof (int)),
+        MethodCallTransformerUtility.GetInstanceMethod (
+            typeof (DateOnly),
+            "AddYears",
+            typeof (int))
```

Comment:
I'm surprised about the linebreaks since we don't need them with DateTime MethodCallTransformer. It's okay if you want to keep them, though personally, I find it easier to scan the list of methods without the line breaks.